File: src/TypeFormatter/SymbolTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import { SymbolType } from "../Type/SymbolType.js";
import type { BaseType } from "../Type/BaseType.js";

export class SymbolTypeFormatter implements SubTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof SymbolType;
    }
    public getDefinition(type: SymbolType): Definition {
        return {};
    }
    public getChildren(type: SymbolType): BaseType[] {
        return [];
    }
}

File: internal/formatter/symbol_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// SymbolTypeFormatter mirrors src/TypeFormatter/SymbolTypeFormatter.ts.
type SymbolTypeFormatter struct{}

func NewSymbolTypeFormatter() *SymbolTypeFormatter { return &SymbolTypeFormatter{} }

func (f *SymbolTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.SymbolType)
	return ok
}

func (f *SymbolTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return &schema.Definition{}
}

func (f *SymbolTypeFormatter) GetChildren(t types.Type) []types.Type { return nil }
